oeps/best-practices/oep-0051-bp-conventional-commits.rst

```diff
@@ -207,11 +211,13 @@ Tooling
 
 One of the advantages of formalized commit messages is using them as input to tooling and conformance checkers.  We will investigate tooling in the future, and are making no recommendations now.
 
-
+As we trial conformance checking, one open question is whether or not "squash" commits should pass or fail commit linting. See an initial discussion of `linting squash commits in this PR comment thread<https://github.com/edx/open-edx-proposals/pull/254#discussion_r741051683>`__.
 
```

Comment:
I think it needed to be added to here, the "squash".
 https://github.com/edx/edx-lint/blob/3584abab13b3d5d1f93301119da3aded81e51bb6/commitlint.config.js#L62-L67 
 Hence commitlint test is failing now and since @robrap  is using it

Reply:
May be to change here as well: https://github.com/edx/edx-lint/blob/3584abab13b3d5d1f93301119da3aded81e51bb6/edx_lint/files/commitlint.config.js#L5-L10

Reply:
Our pre-merge check validates that a PR is ready to merge to the main branch, as such (I contend in the linked thread) no tooling change is needed as the current commitlint check does not accept this prefix and flags it as not-for-merging unless/until it is squashed away.

If we were using a pre-commit check, that would need to allow `squash`, but the pre-merge check still (I argue) should not.

Reply:
If commitlint were made to be a blocking check, then I would need to squash in order to merge -- but then that would also invalidate all the other checks and I'd have to wait another 20-30 minutes for all of those to run before I could actually merge.

If commitlint is not blocking... then I can just squash-merge and fix the commit title as I do it.

Reply:
Moving commitlint to a blocking check is a much bigger discussion with much bigger consequences

Reply:
We've intentionally split the tooling question from this PR. Please see https://github.com/edx/open-edx-proposals/pull/254#discussion_r741051683 if you wish to add to the debate on how the new "squash" label should should interact with linting.
1. If you have new pros/cons for either option, feel free to add a comment on the other thread.
2. If you simply want to vote for the option you prefer, feel free to add a comment on the other thread.

**Note:** Before this change, I personally was using "temp" and "fix" instead of "squash", and thought these were the correct labels to use. With this change, I would stop doing that and I think the intention around my commits would be more clear. I still have questions about tooling, but I think either way we go, this feels like a step in the right direction.

Reply:
Sorry, yes, I got turned around—deferring the question of linting behavior sounds good. Having a special "blessed illegal value" seems fine. :-)

Reply:
After further discussion:
1. It was too difficult to separate these concerns, because after reading about "squash" commits, it would be nice to provide engineers and expectation if this really would still fail their builds, which would be odd as compared to all the other types.
2. It may be palatable to everyone to make "squash" commits warn rather than error, but not fail the build. See https://github.com/edx/open-edx-proposals/pull/254/files#r743981426 for discussion of that change in action.

Reply:
Added more details to above comment, so switch threads to stay current. :)